core/src/main/java/tc/oc/pgm/loot/LootableModule.java

```diff
@@ -0,0 +1,125 @@
+import tc.oc.pgm.filters.FilterMatchModule;
+import tc.oc.pgm.filters.matcher.StaticFilter;
+import tc.oc.pgm.filters.parse.DynamicFilterValidation;
+import tc.oc.pgm.itemmeta.ItemModifyMatchModule;
+import tc.oc.pgm.util.TimeUtils;
+import tc.oc.pgm.util.compose.Composition;
+import tc.oc.pgm.util.compose.CompositionParser;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class LootableModule implements MapModule<LootableMatchModule> {
+
+  private static final Function<MapFactory, CompositionParser<ItemStack>> LOOT_PARSER_FACTORY =
+      factory ->
+          new CompositionParser<ItemStack>(factory) {
+            @Override
+            public ItemStack parseUnit(Element element) throws InvalidXMLException {
+              return this.factory.getKits().parseItemStack(element);
+            }
+          };
+
+  private final java.util.List<FillerDefinition> fillers;
+  private final List<Cache> caches;
+
+  public LootableModule(List<FillerDefinition> fillers, List<Cache> caches) {
+    this.fillers = fillers;
+    this.caches = caches;
+  }
+
+  @Nullable
+  @Override
+  public LootableMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new LootableMatchModule(match.getLogger(), match, this.fillers, this.caches);
+  }
+
+  @Nullable
+  @Override
+  public Collection<Class<? extends MatchModule>> getHardDependencies() {
+    return ImmutableList.of(FilterMatchModule.class);
+  }
+
+  @Nullable
+  @Override
+  public Collection<Class<? extends MatchModule>> getWeakDependencies() {
+    return ImmutableList.of(ItemModifyMatchModule.class);
+  }
+
+  public static class Factory implements MapModuleFactory<LootableModule> {
+
+    @Nullable
+    @Override
+    public LootableModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      List<FillerDefinition> fillers = new ArrayList<>();
+      List<Cache> caches = new ArrayList<>();
+      CompositionParser<ItemStack> lootParser = LOOT_PARSER_FACTORY.apply(factory);
+      Map<String, Composition<ItemStack>> loots = new HashMap<>();
+
+      for (Element loot : XMLUtils.flattenElements(doc.getRootElement(), "lootables", "loot")) {
+        loots.put(loot.getAttributeValue("id"), lootParser.parseElement(loot));
+      }
+
+      for (Element filler : XMLUtils.flattenElements(doc.getRootElement(), "lootables", "fill")) {
+        Composition<ItemStack> loot = loots.get(filler.getAttributeValue("loot"));
+        Filter fillableFilter =
+            factory.getFilters().parseFilterProperty(filler, "filter", StaticFilter.ALLOW);
+        Duration refillInterval =
+            XMLUtils.parseDuration(
+                filler.getAttribute("refill-interval"), TimeUtils.INFINITE_DURATION);
+        @Nullable
+        Filter refillTrigger =
+            factory
+                .getFilters()
+                .parseProperty(filler, "refill-trigger", DynamicFilterValidation.MATCH);
```

Comment:
as mentioned earlier, could make this non-nullable and simply make the default a StaticFilter.DENY/never